### PR TITLE
feature #710:Implemented endpoint for receiving groups with connected…

### LINKED ISSUE #710: Implement endpoint for receiving groups with connected telegram groups
Implement endpoint for receiving groups with connected telegram groups

Needed fields: id, telegramId, source

### PR DESCRIPTION
**Pull request by illiakroshka**
… telegram groups

- implemented for receiving groups with connected telegram groups
- add swagger documentation

closes #710

### CODE REVIEW

File: src/v2/api/services/TelegramGroupService.ts [hunk 1]
```diff
@@ -4,10 +4,14 @@ import { CreateTelegramGroupDTO } from '../dtos/CreateTelegramGroupDTO';
 import { UpdateTelegramGroupDTO } from '../dtos/UpdateTelegramGroupDTO';
 import { DataNotFoundException } from '../../utils/exceptions/DataNotFoundException';
 import { AlreadyExistException } from '../../utils/exceptions/AlreadyExistException';
+import { GroupRepository } from '../../database/repositories/GroupRepository';
+
```
Comment (PavloVlasiuk) [inline src/v2/api/services/TelegramGroupService.ts:8]: remove one line here

File: src/v2/api/services/TelegramGroupService.ts [hunk 2]
```diff
@@ -43,4 +47,8 @@ export class TelegramGroupService {
   async getGroupByTelegramId (telegramId: bigint) {
     return this.telegramGroupRepository.findByTelegramId(telegramId);
   }
+
+  async findGroupsWithTelegramGroups () {
```
Comment (PavloVlasiuk) [inline src/v2/api/services/TelegramGroupService.ts:51]: i would name this method getGroupsWithTelegramGroups

File: src/v2/api/controllers/TelegramGroupController.ts [hunk 3]
```diff
@@ -180,4 +181,19 @@ export class TelegramGroupController {
     const telegramGroups = await this.telegramGroupService.getGroupByTelegramId(telegramId);
     return this.telegramGroupMapper.getGroupsByTelegramId(telegramGroups);
   }
+
```
Comment (PavloVlasiuk) [inline src/v2/api/controllers/TelegramGroupController.ts:184]: add ApiBearerAuth here

File: src/v2/mappers/TelegramGroupMapper.ts [hunk 4]
```diff
@@ -25,4 +26,16 @@ export class TelegramGroupMapper {
       })),
     };
   }
+
+  getGroupsWithTelegramGroups (groups: DbGroupWithTelegramGroups[]) {
+    return {
+      groups: groups.map((telegramGroup) => ({
```
Comment (PavloVlasiuk) [inline src/v2/mappers/TelegramGroupMapper.ts:32]: rename the argument of the first map to group and the the argument of the second map to telegramGroup

**Review by PavloVlasiuk — CHANGES_REQUESTED** [review]

**Review by PavloVlasiuk — APPROVED** [review]

File: src/v2/mappers/TelegramGroupMapper.ts [hunk 5]
```diff
@@ -26,4 +27,17 @@ export class TelegramGroupMapper {
...
     };
   }
+
+  getGroupsWithTelegramGroups (groups: DbGroupWithTelegramGroups[]) {
+    return {
+      groups: groups.map((group) => ({
+        id: group.id,
+        telegramGroups: group.telegramGroups.map((telegramGroup) => ({
+          telegramId: telegramGroup.telegramId,
+          theadId: telegramGroup.threadId,
```
Comment (Ksrebniukk) [inline src/v2/mappers/TelegramGroupMapper.ts:37]: replace "theadId" with "threadId"

**Review by Ksrebniukk — APPROVED** [review]

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**